tutorials/early_stopping/early_stopping.ipynb

```diff
@@ -0,0 +1,1067 @@
+    "- The `min_progression` parameter specifies that trials should only be considered for stopping if the latest progression value is greater than this threshold.\n",
+    "- The `min_curves` parameter specifies the minimum number of completed curves before early stopping will be considered. This should be larger than zero if `normalize_progression` is used. In general, we want a few completed curves to have a baseline for comparison."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "949e8ab5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "percentile_early_stopping_strategy = PercentileEarlyStoppingStrategy(\n",
+    "    # stop if in bottom 50% of runs at the same progression\n",
+    "    percentile_threshold=50,\n",
+    "    # the trial must have passed `min_progression` steps before early stopping is initiated\n",
+    "    # note that we are using `normalize_progressions`, so this is on a scale of [0, 1]\n",
+    "    min_progression=0.3,\n",
+    "    # there must be `min_curves` completed trials and `min_curves` trials reporting data in\n",
+    "    # order for early stopping to be applicable\n",
+    "    min_curves=3,\n",
+    "    # specify, e.g., [0, 1] if the first two trials should never be stopped\n",
+    "    trial_indices_to_ignore=None,\n",
+    "    # check for new data every 15 seconds\n",
+    "    seconds_between_polls=15,\n",
+    "    normalize_progressions=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2665ca93",
+   "metadata": {},
+   "source": [
+    "## Creating the Ax Experiment\n",
+    "\n",
+    "In Ax, the Experiment object is the object that stores all the information about the problem setup."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "12849b31",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "experiment = Experiment(\n",
+    "    name=\"torchx_mnist\",\n",
+    "    search_space=search_space,\n",
+    "    optimization_config=opt_config,\n",
+    "    runner=ax_runner,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49a4ed0e",
+   "metadata": {},
+   "source": [
+    "## Choosing the GenerationStrategy\n",
+    "\n",
+    "A [GenerationStrategy](https://ax.dev/api/modelbridge.html#ax.modelbridge.generation_strategy.GenerationStrategy)_\n",
+    "is the abstract representation of how we would like to perform the\n",
+    "optimization. While this can be customized (if you’d like to do so, see\n",
+    "[this tutorial](https://ax.dev/tutorials/generation_strategy.html)_),\n",
+    "in most cases Ax can automatically determine an appropriate strategy\n",
+    "based on the search space, optimization config, and the total number\n",
+    "of trials we want to run.\n",
+    "\n",
+    "Typically, Ax chooses to evaluate a number of random configurations\n",
+    "before starting a model-based Bayesian Optimization strategy."
```

Comment:
We should probably talk about (or at least hint at) the fact that generation strategy and early stopping strategy are interdependent, and that in an ideal world we'd consider this as one big problem, but that this generally works quite well in practice and is a lot simpler to set up.